subsys/fs/Kconfig.fatfs

```diff
@@ -211,6 +212,13 @@ config FS_FATFS_WINDOW_ALIGNMENT
 	  that, in worst scenario, value provided here may cause FATFS
 	  structure to have size of twice the value.
 
+config FS_FATFS_REENTRANT
+	bool "FatFs reentrant"
+	help
```

Comment:
Would a `depends on !FS_FATFS_LFN_MODE_BSS` make sense here? Without, it's really easy to enable LFN in its default configuration, enable reentrancy, and not realize that it's actually still not thread safe. With the `depends`, there would at least be a warning because `CONFIG_FS_FATFS_REENTRANT` can't be selected.

Reply:
Yeah, we should probably add that.

Reply:
Good point!
I tested it locally with `depends on !FS_FATFS_LFN_MODE_BSS`:
I added the `CONFIG_FS_FATFS_LFN=y` in the testcase.yaml for the reentrant test in the `extra_configs`, but the test did not fail until I set the mode manually with `FS_FATFS_LFN_MODE_BSS=y`. I also tested it by adding the `depends on !FS_FATFS_REENTRANT` to `FS_FATFS_LFN_MODE_BSS` and had the same behaviour.
The default mode is probably the most critical choise to warn about which still won't work.
Do you know more about this behaviour?

Reply:
I added `depends on !FS_FATFS_REENTRANT` to `FS_FATFS_LFN_MODE_BSS` because `FS_FATFS_LFN_MODE_BSS` is not always visible and therefore I think this way round it is cleaner.